packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -497,15 +497,15 @@ function renderHTMLDocument (html: NuxtRenderHTMLContext) {
 
 async function renderInlineStyles (usedModules: Set<string> | string[]): Promise<Style[]> {
   const styleMap = await getSSRStyles()
-  const inlinedStyles = new Set<string>()
+  const inlinedStyles: Style[] = []
   for (const mod of usedModules) {
     if (mod in styleMap) {
       for (const style of await styleMap[mod]()) {
-        inlinedStyles.add(style)
+        inlinedStyles.push({ key: mod, innerHTML: style })
       }
     }
   }
-  return Array.from(inlinedStyles).map(style => ({ innerHTML: style }))
+  return inlinedStyles
```

Comment:
if performance is the main concern it's probably faster to dedupe in this function?

```ts
async function renderInlineStyles (usedModules: Set<string> | string[]): Promise<Style[]> {
  const styleMap = await getSSRStyles()
  const inlinedStyles: Record<string, string> = {}
  for (const mod of usedModules) {
    if (mod in styleMap) {
      for (const style of await styleMap[mod]()) {
        inlinedStyles[mod] = style
      }
    }
  }
  return Object.values(inlinedStyles).map(style => ({ innerHTML: style }))
}
```

Reply:
No, those styles are never hydrated 👍